### PR TITLE
Add plan lifecycle rule

### LINKED ISSUE #154: Add plan-lifecycle rule: Plan → ADR lifecycle for non-trivial features
## Summary

Add `.claude/rules/common/plan-lifecycle.md` to codify the Plan → ADR lifecycle so Ballast can install it in any project via `ballast init`, the same way linting, testing, and git-hook rules are distributed.

## Problem

The plan/ADR lifecycle currently lives in `plans/plan-lifecycle.md` in this repo only. It is not distributed to other projects via Ballast. Agents working in other repos have no guidance on how to create plans, maintain them during implementation, or graduate them to ADRs at merge time.

## Goal

Make the plan → ADR lifecycle a first-class Ballast rule available to all projects.

## Relationship to existing rules

| Rule | Concern |
|------|---------|
| `tasks-todo.md` | The TODO scratchpad file and PR triage gate |
| `tasks-task-system.md` | Durable issue tracker (GitHub/Jira/Linear) and MCP setup |
| `plan-lifecycle.md` *(new)* | Plan → ADR lifecycle for non-trivial features |

The new rule should defer to `tasks-todo.md` as authoritative for `tasks/todo.md` behavior (the parking lot for discovered out-of-scope work) rather than re-specifying it. The graduation gate — all open TODO items resolved before merge — reinforces what `tasks-todo.md` already defines.

### 1. Create `.claude/rules/common/plan-lifecycle.md`

This rule is language-agnostic and belongs in `common/` only — no per-language duplicates needed.

The rule must cover:

#### When to create a plan

Create a plan when:
- The change touches more than 2 files
- You are unsure of the approach
- The feature takes more than one session to complete
- The work involves architectural decisions

Skip a plan when:
- Single-file fix (typo, log line, rename)
- The entire change fits in one sentence

#### Directory structure

```
<project-root>/
├── plans/
│   ├── README.md
│   └── plan-<feature-name>.md
├── tasks/
│   └── todo.md                   ← discovered out-of-scope work, cleared at graduation
└── adr/
    ├── README.md
    └── NNN-<decision-title>.md
```

#### Plan template

File: `plans/plan-<feature-name>.md` (kebab-case, specific: `plan-oauth-google.md` not `plan-auth.md`)

```markdown
# Plan: <Feature Name>

**Status:** In Progress
**Branch:** <branch-name>
**Created:** YYYY-MM-DD
**Related ADRs:** _(link any relevant existing ADRs)_

## Problem

What are we solving and why does it matter now?

## Approach

The chosen solution in plain language. What will change and how.

## Files Affected

- `src/...` — reason
- `tests/...` — reason

## Phases

- [ ] Phase 1: Explore and confirm approach
- [ ] Phase 2: Core implementation
- [ ] Phase 3: Tests and edge cases
- [ ] Phase 4: Documentation and cleanup

## Verification

How will we know this works? Commands, tests, or checks to run.

## Alternatives Rejected

| Option | Why rejected |
|--------|-------------|
| ...    | ...         |

## Open Questions

Things still to resolve. Remove entries as they are answered.

## Change Log

| Date | Change |
|------|--------|
| YYYY-MM-DD | Plan created |
```

After creating the plan, commit it and update `plans/README.md`.

#### Maintaining the plan during implementation

- Check off phases as they complete
- If approach changes, update **Approach** and note it in **Change Log**
- Commit plan updates alongside code changes — not as a separate commit at the end
- At the start of each session, read the plan to restore context
- When you discover work that is real but out of scope for this feature, add it to `tasks/todo.md` (see `tasks-todo.md` rule) rather than expanding the plan

#### Graduation: graduating the plan to an ADR

When the feature is ready to merge, use the trigger phrase:

> "Graduate `plans/plan-<feature-name>.md` to an ADR"

Graduation steps:
1. **Check `tasks/todo.md`** for any incomplete items (`- [ ]`) added during this feature
2. **For each incomplete item:** create a task system work item, then update the line to `- [x] TASK-NNN: <description>`
3. Determine the next ADR number from `adr/README.md`
4. Create `adr/NNN-<decision-title>.md` from the plan content
5. Update `adr/README.md` with the new row
6. Remove `plans/plan-<feature-name>.md`
7. Update `plans/README.md` to remove the entry
8. Commit: `docs: graduate plan-<feature-name> to ADR-NNN`

> Graduation is blocked until all items in `tasks/todo.md` for this feature are either checked off or have a task system reference.

#### ADR template

```markdown
# ADR-NNN: <Decision Title>

**Status:** Accepted
**Date:** YYYY-MM-DD
**Branch:** <branch-name>
**PR:** #<number>
**Supersedes:** _(ADR-NNN if replacing an earlier decision)_
**Superseded by:** _(leave blank)_

## Context

Why did this decision need to be made?

## Decision

What was chosen and why.

## Alternatives Considered

| Option | Reason not chosen |
|--------|------------------|
| ...    | ...              |

## Consequences

### Positive
- What becomes easier

### Negative or trade-offs
- What becomes harder or what we gave up

## Implementation Notes

Key details future readers should know.

## Verification

How the decision was validated.

## Lessons Learned

What would you do differently? What worked better than expected?
```

#### ADR management rules

| Rule | Detail |
|------|--------|
| **Never delete** | Mark superseded, create a new ADR |
| **Sequential numbering** | Zero-padded three digits: `001`, `002`, `003` |
| **One decision per ADR** | Don't bundle unrelated decisions |
| **Status values** | `Accepted` · `Deprecated` · `Superseded` |

#### Quick reference

| Situation | Action |
|-----------|--------|
| Starting a feature | Create `plans/plan-<name>.md`, commit it |
| New session on existing feature | "Continue implementing `plans/plan-<name>.md`" |
| Approach changed | Update plan + Change Log, commit with code |
| Phase complete | Check off in plan, commit |
| Discovered out-of-scope work | Add to `tasks/todo.md`, commit alongside current change |
| Ready to merge | "Graduate `plans/plan-<name>.md` to an ADR" |
| Graduation finds open todo items | Create task system issues, add references to `tasks/todo.md`, then proceed |
| Decision reversed later | Mark ADR `Superseded`, create new ADR |
| Small single-file fix | Skip the plan entirely |

### 2. Wire into Ballast installation

Ensure `ballast init` can distribute this rule to target repos.

### 3. Update `CLAUDE.md` installed rules list

Add the rule entry alongside the existing tasks rules:

```markdown
- `.claude/rules/common/plan-lifecycle.md` — Rules for common/plan-lifecycle
```

## Acceptance criteria

- `.claude/rules/common/plan-lifecycle.md` exists and covers: when to create a plan, the plan template, how to maintain it during implementation, the graduation ceremony (plan → ADR), the ADR template, ADR management rules, and the quick reference table
- The rule defers to `tasks-todo.md` for `tasks/todo.md` behaviour rather than duplicating it
- Task system references are generic (not Jira-specific)
- The rule is installable via Ballast and listed in `CLAUDE.md`

### PR DESCRIPTION
**Pull request by markcallen**
## Summary

- Adds the common `plan-lifecycle` agent rule for creating plans, maintaining them during implementation, and graduating completed plans to ADRs.
- Registers `plan-lifecycle` across the wrapper and TypeScript backend, updates `.rulesrc.json`, and refreshes checked-in Codex/Claude generated outputs plus Python/Go package mirrors.
- Adds PRD requirements and generated-content/support-file tests for issue #154.

## Validation

- [x] Tests or checks run: `pnpm --dir packages/ballast-typescript test`; `pnpm --dir packages/ballast-typescript run lint`; `go test ./...` in `cli/ballast`; `git diff --check`
- [x] Docs updated or not needed: `PRD.md`, `AGENTS.md`, and `CLAUDE.md` updated
- [x] Generated files updated or not needed: `.claude/rules/common/plan-lifecycle.md`, `.codex/rules/common/plan-lifecycle.md`, `.codex/rules/typescript/typescript-plan-lifecycle.md`, and package mirrors refreshed

## Agent Notes

- Related issue/PRD: Closes #154; `PRD.md` Plan Lifecycle Rule Distribution
- Risk level: Low; generated guidance and registry/config update only
- Follow-up needed: None

### CODE REVIEW

File: tasks/todo.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # Tasks
 
+- [x] Confirm issue #154 scope, operating mode, and governing PRD requirements.
```
[the remark this answers is not in the document]
Comment (markcallen) [reply tasks/todo.md:3]: Fixed in 51dfc9b by standardizing the generated task guidance and plan-lifecycle guidance on the repo canonical lowercase `tasks/todo.md` path, avoiding split TODO files on case-sensitive filesystems.

File: agents/common/plan-lifecycle/templates/cursor-frontmatter.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+description: 'Plan lifecycle - create, maintain, and graduate plans to ADRs'
+alwaysApply: false
+globs:
+  - 'plans/**/*.md'
+  - 'adr/**/*.md'
+  - 'tasks/TODO.md'
```
[the remark this answers is not in the document]
Comment (markcallen) [reply agents/common/plan-lifecycle/templates/cursor-frontmatter.yaml]: Fixed in 51dfc9b by changing the plan-lifecycle Cursor glob from `tasks/TODO.md` to `tasks/todo.md` and syncing the package mirrors/generated outputs.

File: PRD.md [hunk 3]
```diff
@@ -136,6 +136,32 @@ Ballast-generated rule files for persistent agent context have accumulated large
...
+3. The rule must include a plan template with status, branch, created date, related ADRs, problem, approach, files affected, phases, verification, rejected alternatives, open questions, and change log.
+4. The rule must describe maintaining the plan during implementation, including checking off phases, updating the approach when it changes, committing plan updates with related code, reading the plan at session start, and routing out-of-scope work to the TODO rule.
+5. The rule must define the graduation trigger and steps for turning a plan into an ADR, including checking `tasks/TODO.md`, creating generic task-system work items for incomplete TODOs, assigning the next ADR number, creating the ADR, updating indexes, removing the plan, and committing the graduation.
+6. The rule must include an ADR template and ADR management rules for status values, sequential numbering, one decision per ADR, and never deleting ADR history.
+7. The rule must defer to the existing `tasks-todo` rule for `tasks/TODO.md` behavior instead of restating that rule in detail.
```
[the remark this answers is not in the document]
Comment (markcallen) [reply PRD.md]: Fixed in 51dfc9b by updating the PRD requirements and acceptance criteria to reference lowercase `tasks/todo.md`, matching the repo file and generated guidance.

File: agents/common/plan-lifecycle/content.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+Use this project-root structure:
+
+```text
+<project-root>/
++-- plans/
+|   +-- README.md
+|   +-- plan-<feature-name>.md
++-- tasks/
+|   +-- TODO.md
```
[the remark this answers is not in the document]
Comment (markcallen) [reply agents/common/plan-lifecycle/content.md]: Fixed in 51dfc9b by updating the plan-lifecycle rule content and the existing tasks TODO/task-system rule sources to use lowercase `tasks/todo.md` consistently, then syncing mirrors and generated outputs.

File: .codex/rules/common/tasks-todo.md [hunk 5]
```diff
@@ -2,28 +2,28 @@
 
 These rules are intended for Codex (CLI and app).
 
-Manage `tasks/TODO.md` during branch work. Triage all unchecked items before creating a PR.
+Manage `tasks/todo.md` during branch work. Triage all unchecked items before creating a PR.
```
[the remark this answers is not in the document]
Comment (markcallen) [reply .codex/rules/common/tasks-todo.md:6]: Fixed in 6a34efb by updating the tracked root Codex task outputs `.codex/rules/tasks-todo.md` and `.codex/rules/tasks-task-system.md` to use lowercase `tasks/todo.md`, matching the common/type-scoped generated outputs and repository file.
